### ISSUE TITLE
Installation issue: julia-1.10.2

### ISSUE THREAD

**Issue by thalunil** [LLNL/spack#44555]
### Steps to reproduce the issue

```console
spack spec -I julia@1.10.2
Input spec
--------------------------------
 -   julia@1.10.2

Concretized
--------------------------------
 -   julia@1.10.2%gcc@13.2.0+openlibm+precompile build_system=makefile patches=00569f4,7f0388b arch=linux-rocky9-x86_64_v2
[+]      ^curl@8.7.1%gcc@13.2.0~gssapi~ldap~libidn2~librtmp~libssh+libssh2+nghttp2 build_system=autotools libs=shared,static tls=mbedtls,openssl arch=linux-rocky9-x86_64_v2
[+]          ^openssl@3.3.0%gcc@13.2.0~docs+shared build_system=generic certs=mozilla arch=linux-rocky9-x86_64_v2
[+]              ^ca-certificates-mozilla@2023-05-30%gcc@13.2.0 build_system=generic arch=linux-rocky9-x86_64_v2
[+]          ^pkgconf@2.2.0%gcc@11.4.1 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]      ^dsfmt@2.2.5%gcc@13.2.0 build_system=makefile patches=b79624c arch=linux-rocky9-x86_64_v2
[+]      ^gcc-runtime@13.2.0%gcc@13.2.0 build_system=generic arch=linux-rocky9-x86_64_v2
[e]      ^glibc@2.34%gcc@11.4.1 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]      ^gmake@4.4.1%gcc@11.4.1~guile build_system=generic arch=linux-rocky9-x86_64_v2
[+]          ^gcc-runtime@11.4.1%gcc@11.4.1 build_system=generic arch=linux-rocky9-x86_64_v2
[+]      ^gmp@6.2.1%gcc@11.4.1+cxx build_system=autotools libs=shared,static patches=69ad2e2 arch=linux-rocky9-x86_64_v2
[+]          ^autoconf@2.72%gcc@11.4.1 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]          ^automake@1.16.5%gcc@11.4.1 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]          ^libtool@2.4.7%gcc@11.4.1 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]              ^findutils@4.9.0%gcc@11.4.1 build_system=autotools patches=440b954 arch=linux-rocky9-x86_64_v2
[+]          ^m4@1.4.19%gcc@11.4.1+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-rocky9-x86_64_v2
[+]              ^libsigsegv@2.14%gcc@11.4.1 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]      ^libblastrampoline@5.8.0%gcc@13.2.0 build_system=makefile arch=linux-rocky9-x86_64_v2
[+]      ^libgit2@1.6.4%gcc@13.2.0~curl~ipo+mmap+ssh build_system=cmake build_type=Release generator=make https=system arch=linux-rocky9-x86_64_v2
[+]          ^cmake@3.27.9%gcc@13.2.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-rocky9-x86_64_v2
[+]          ^pcre@8.45%gcc@13.2.0~jit+multibyte+pic+shared+static+utf build_system=autotools arch=linux-rocky9-x86_64_v2
[+]      ^libssh2@1.11.0%gcc@13.2.0+shared build_system=autotools crypto=mbedtls patches=011d926 arch=linux-rocky9-x86_64_v2
[+]          ^xz@5.4.6%gcc@11.4.1~pic build_system=autotools libs=shared,static arch=linux-rocky9-x86_64_v2
[+]      ^libunwind@1.6.2%gcc@13.2.0~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs~pic+tests+weak_backtrace~xz~zlib build_system=autotools components=none libs=shared,static arch=linux-rocky9-x86_64_v2
[+]      ^libuv-julia@1.44.3%gcc@13.2.0 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]      ^libwhich@1.1.0%gcc@13.2.0 build_system=generic arch=linux-rocky9-x86_64_v2
[+]      ^llvm@15.0.7%gcc@13.2.0+clang~cuda~flang+gold~ipo+libomptarget~libomptarget_debug+link_llvm_dylib+lld+lldb+llvm_dylib+lua~mlir+polly~python~split_dwarf~z3~zstd build_system=cmake build_type=Release compiler-rt=runtime generator=ninja libcxx=runtime libunwind=none openmp=runtime patches=25bc503,25cdc02,3dc9bc1,8248141,90f3d78 shlib_symbol_version=JL_LLVM_15.0 targets=all,amdgpu,bpf,nvptx,webassembly version_suffix=jl arch=linux-rocky9-x86_64_v2
[+]          ^binutils@2.42%gcc@13.2.0~gas+gold~gprofng+headers~interwork+ld~libiberty~lto~nls~pgo+plugins build_system=autotools compress_debug_sections=zlib libs=shared,static arch=linux-rocky9-x86_64_v2
[+]              ^zstd@1.5.6%gcc@11.4.1+programs build_system=makefile compression=none libs=shared,static arch=linux-rocky9-x86_64_v2
[+]          ^elfutils@0.190%gcc@13.2.0~debuginfod+exeprefix+nls build_system=autotools arch=linux-rocky9-x86_64_v2
[+]          ^hwloc@2.9.1%gcc@13.2.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-rocky9-x86_64_v2
[+]              ^libpciaccess@0.17%gcc@13.2.0 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]                  ^util-macros@1.19.3%gcc@13.2.0 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]          ^libedit@3.1-20230828%gcc@13.2.0 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]          ^libffi@3.4.6%gcc@13.2.0 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]          ^libxml2@2.10.3%gcc@11.4.1+pic~python+shared build_system=autotools arch=linux-rocky9-x86_64_v2
[+]          ^lua@5.3.6%gcc@13.2.0~pcfile+shared build_system=makefile fetcher=curl arch=linux-rocky9-x86_64_v2
[+]              ^unzip@6.0%gcc@13.2.0 build_system=makefile patches=f6f6236 arch=linux-rocky9-x86_64_v2
[+]          ^ncurses@6.5%gcc@11.4.1~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-rocky9-x86_64_v2
[+]          ^ninja@1.11.1%gcc@13.2.0+re2c build_system=generic arch=linux-rocky9-x86_64_v2
[+]              ^re2c@2.2%gcc@13.2.0 build_system=generic arch=linux-rocky9-x86_64_v2
[+]          ^perl-data-dumper@2.173%gcc@13.2.0 build_system=perl arch=linux-rocky9-x86_64_v2
[+]          ^swig@4.0.2-fortran%gcc@13.2.0 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]              ^bison@3.8.2%gcc@13.2.0~color build_system=autotools arch=linux-rocky9-x86_64_v2
[+]      ^mbedtls@2.28.2%gcc@13.2.0+pic build_system=makefile build_type=Release libs=shared,static arch=linux-rocky9-x86_64_v2
[+]      ^mpfr@4.2.1%gcc@11.4.1 build_system=autotools libs=shared,static arch=linux-rocky9-x86_64_v2
[+]          ^autoconf-archive@2023.02.20%gcc@11.4.1 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]          ^texinfo@7.0.3%gcc@11.4.1 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]      ^nghttp2@1.52.0%gcc@13.2.0 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]          ^diffutils@3.10%gcc@11.4.1 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]      ^openblas@0.3.26%gcc@13.2.0~bignuma~consistent_fpcsr+dynamic_dispatch+fortran+ilp64+locking+pic+shared build_system=makefile symbol_suffix=64_ threads=none arch=linux-rocky9-x86_64_v2
[+]      ^openlibm@0.8.1%gcc@13.2.0 build_system=makefile arch=linux-rocky9-x86_64_v2
[+]      ^p7zip@17.05%gcc@13.2.0 build_system=makefile arch=linux-rocky9-x86_64_v2
[+]      ^patchelf@0.17.2%gcc@13.2.0 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]      ^pcre2@10.43%gcc@13.2.0~jit+multibyte build_system=autotools arch=linux-rocky9-x86_64_v2
[+]      ^perl@5.38.0%gcc@11.4.1+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-rocky9-x86_64_v2
[+]          ^berkeley-db@18.1.40%gcc@11.4.1+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-rocky9-x86_64_v2
[+]          ^bzip2@1.0.8%gcc@11.4.1~debug~pic+shared build_system=generic arch=linux-rocky9-x86_64_v2
[+]          ^gdbm@1.23%gcc@11.4.1 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]      ^python@3.11.7%gcc@13.2.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060 arch=linux-rocky9-x86_64_v2
[+]          ^expat@2.6.2%gcc@13.2.0+libbsd build_system=autotools arch=linux-rocky9-x86_64_v2
[+]              ^libbsd@0.12.1%gcc@13.2.0 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]                  ^libmd@1.0.4%gcc@13.2.0 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]          ^gettext@0.22.5%gcc@11.4.1+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-rocky9-x86_64_v2
[+]              ^tar@1.34%gcc@11.4.1 build_system=autotools zip=pigz arch=linux-rocky9-x86_64_v2
[+]                  ^pigz@2.8%gcc@11.4.1 build_system=makefile arch=linux-rocky9-x86_64_v2
[+]          ^libxcrypt@4.4.35%gcc@13.2.0~obsolete_api build_system=autotools patches=4885da3 arch=linux-rocky9-x86_64_v2
[+]          ^readline@8.2%gcc@11.4.1 build_system=autotools patches=bbf97f1 arch=linux-rocky9-x86_64_v2
[+]          ^sqlite@3.43.2%gcc@13.2.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-rocky9-x86_64_v2
[+]          ^util-linux-uuid@2.38.1%gcc@13.2.0 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]      ^suite-sparse@7.2.1%gcc@13.2.0~cuda~graphblas~openmp+pic build_system=generic arch=linux-rocky9-x86_64_v2
[+]          ^metis@5.1.0%gcc@13.2.0~gdb~int64~ipo~real64+shared build_system=cmake build_type=Release generator=make patches=4991da9,93a7903,b1225da arch=linux-rocky9-x86_64_v2
[+]      ^utf8proc@2.8.0%gcc@13.2.0~ipo~shared build_system=cmake build_type=Release generator=make arch=linux-rocky9-x86_64_v2
[+]      ^which@2.21%gcc@13.2.0 build_system=autotools arch=linux-rocky9-x86_64_v2
[+]      ^zlib-ng@2.1.6%gcc@11.4.1+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-rocky9-x86_64_v2

```

### Error message

<details><summary>Error message</summary>

<pre>
==> Installing julia-1.10.2-nfm3ltzpfw7ijwzi6lkchpinepmtx7be [59/59]
==> No binary for julia-1.10.2-nfm3ltzpfw7ijwzi6lkchpinepmtx7be found: installing from source
==> Using cached archive: /usr/site/spackinstall/SIH9q/var/spack/cache/_source-cache/archive/e3/e3d20c02975da054aeb18d32ed84c5d760d54d2563e45e25017684a5a105d185.tar.gz
==> Using cached archive: /usr/site/spackinstall/SIH9q/var/spack/cache/_source-cache/archive/00/00569f40e1845329060a714813e509677949e633a0e833c40a3c70dcf9269cc1
==> Applied patch https://github.com/JuliaLang/julia/commit/55c13d234c1523861b278f7989b1af105ef0e88f.patch?full_index=1
==> Applied patch /usr/site/spackinstall/SIH9q/var/spack/repos/builtin/packages/julia/julia-1.10-rm-suite-sparse-cuda-stubs.patch
==> Ran patch() for julia
==> julia: Executing phase: 'edit'
==> julia: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2: 
    'make' '-j2'

2 errors found in build log:
     132    echo 1 > SHA-2d1f84e6f8417a1a368de48318640d948b023e7a/source-extracted
     133    ln -sf /usr/site/spackinstall/SIH9q/opt/spack/linux-rocky9-x86_64_v2/gcc-13.2.0/mbedtls-2.28.2-k6yaxvzqv7pmt7siubwlkmchsidvzqta/lib/libmbedx509.so.1 /tmp/root/spack-stage/spack-stage-julia-1.10.2-nfm3ltzpfw7ijwzi6lkchpinepmtx7be/spack-src/usr/lib/julia/libm
            bedx509.so
     134    ln -sf /usr/site/spackinstall/SIH9q/opt/spack/linux-rocky9-x86_64_v2/gcc-13.2.0/libssh2-1.11.0-vmskzfmh7ts6ivwb5yahlcqfflcbln73/lib/libssh2.so.1 /tmp/root/spack-stage/spack-stage-julia-1.10.2-nfm3ltzpfw7ijwzi6lkchpinepmtx7be/spack-src/usr/lib/julia/libssh2.
            so
     135    ln -sf /usr/site/spackinstall/SIH9q/opt/spack/linux-rocky9-x86_64_v2/gcc-13.2.0/curl-8.7.1-7an2m6cfsd2o7vtdalbzy6a35ryk2rwe/lib/libcurl.so.4 /tmp/root/spack-stage/spack-stage-julia-1.10.2-nfm3ltzpfw7ijwzi6lkchpinepmtx7be/spack-src/usr/lib/julia/libcurl.so
     136    ln -sf /usr/site/spackinstall/SIH9q/opt/spack/linux-rocky9-x86_64_v2/gcc-13.2.0/libgit2-1.6.4-evgvfjzwzmgwo4rtnjx5m64jf37cxrhn/lib64/libgit2.so.1.6 /tmp/root/spack-stage/spack-stage-julia-1.10.2-nfm3ltzpfw7ijwzi6lkchpinepmtx7be/spack-src/usr/lib/julia/libgi
            t2.so
     137    System library symlink failure: Unable to locate libamd.so on your system!
  >> 138    make[1]: *** [Makefile:274: /tmp/root/spack-stage/spack-stage-julia-1.10.2-nfm3ltzpfw7ijwzi6lkchpinepmtx7be/spack-src/usr/lib/julia/libamd.so] Error 1
     139    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-julia-1.10.2-nfm3ltzpfw7ijwzi6lkchpinepmtx7be/spack-src/base'
  >> 140    make: *** [Makefile:82: julia-base] Error 2
     141    make: *** Waiting for unfinished jobs....  
     142    echo 1 > Pkg-70525539d3157891445abe806962f780f64143cd/build-compiled
     143    echo 1 > Statistics-d147f9253aa0f2f71be9c3fed8d51c2215410408/build-compiled
     144    echo 1 > LibCURL-a65b64f6eabc932f63c2c0a4a5fb5d75f3e688d0/build-compiled
     145    echo 1 > DelimitedFiles-db79c842f95f55b1f8d8037c0d3363ab21cd3b90/build-compiled
     146    echo 1 > Downloads-ead289a7f1be7689738aaea9b12d919424a106ef/build-compiled

See build log for details:
  /tmp/root/spack-stage/spack-stage-julia-1.10.2-nfm3ltzpfw7ijwzi6lkchpinepmtx7be/spack-build-out.txt

</pre></details>

### Information on your system

* **Spack:** 0.22.0 (5fe93fee1eec46a0750bd340198bffcb92ff9eec)
* **Python:** 3.9.18
* **Platform:** linux-rocky9-sandybridge
* **Concretizer:** clingo

### Additional information

[spack-build-out.txt](https://github.com/user-attachments/files/15573601/spack-build-out.txt)
[spack-build-env.txt](https://github.com/user-attachments/files/15573602/spack-build-env.txt)

@giordano  @haampie  @vchuravy

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by thalunil** [comment]
Maybe similar issue #44534

**Comment by giordano** [comment]
> Maybe similar issue #44534

I don't see any similarity at all. Instead, I think this was already fixed by #44214

**Comment by thalunil** [comment]
thanks a lot - i will wait for v0.22.1 to see if this is resolved
